app/notifications/validators.py

```diff
@@ -274,17 +275,62 @@ def check_service_letter_contact_id(service_id, letter_contact_id, notification_
             raise BadRequestError(message=message)
 
 
-def decode_personalisation_files(personalisation_data):
+def validate_personalisation_and_decode_files(json_personalisation):
     errors = []
-    file_keys = [k for k, v in personalisation_data.items() if isinstance(v, dict) and "file" in v]
+    json_personalisation, errors_vars = validate_personalisation(json_personalisation)
+    json_personalisation, errors_num_file = validate_personalisation_num_files(json_personalisation)
+    json_personalisation, errors_files = decode_personalisation_files(json_personalisation)
+    errors.extend(errors_vars)
+    errors.extend(errors_num_file)
+    errors.extend(errors_files)
+    return json_personalisation, errors
+
+
+def validate_personalisation(json_personalisation):
```

Comment:
could make the function name more specific, ie `validate_personalisation_size()` perhaps, similar to how you did for `validate_personalisation_num_files()`

Reply:
Oh yes, I meant to do that. Thanks!